Gem/Code/Source/Automation/ScriptManager.cpp

```diff
@@ -1406,36 +1409,71 @@ namespace AtomSampleViewer
         s_instance->m_scriptOperations.push(AZStd::move(operation));
     }
 
-    bool ScriptManager::PrepareForScreenCapture(const AZStd::string& path, const AZStd::string& envPath)
+    bool ScriptManager::PrepareForScreenCapture(const AZStd::string& imageName)
     {
-        if (!Utils::IsFileUnderFolder(Utils::ResolvePath(path), ScreenshotPaths::GetScreenshotsFolder(true)))
-        {
-            // The main reason we require screenshots to be in a specific folder is to ensure we don't delete or replace some other important file.
-            ReportScriptError(AZStd::string::format(
-                "Screenshots must be captured under the '%s' folder. Attempted to save screenshot to '%s'.",
-                ScreenshotPaths::GetScreenshotsFolder(false).c_str(), path.c_str()));
-
-            return false;
-        }
+        AZStd::string fullFilePath;
+        AZ::Render::FrameCaptureTestRequestBus::BroadcastResult(
+            fullFilePath,
+            &AZ::Render::FrameCaptureTestRequestBus::Events::BuildScreenshotFilePath,
+            imageName, true);
 
         // Delete the file if it already exists because if the screen capture fails, we don't want to do a screenshot comparison test using an old screenshot.
-        if (AZ::IO::LocalFileIO::GetInstance()->Exists(path.c_str()) && !AZ::IO::LocalFileIO::GetInstance()->Remove(path.c_str()))
+        if (AZ::IO::LocalFileIO::GetInstance()->Exists(fullFilePath.c_str()) && !AZ::IO::LocalFileIO::GetInstance()->Remove(fullFilePath.c_str()))
         {
-            ReportScriptError(AZStd::string::format("Failed to delete existing screenshot file '%s'.", path.c_str()));
+            ReportScriptError(AZStd::string::format("Failed to delete existing screenshot file '%s'.", fullFilePath.c_str()));
             return false;
         }
 
-        s_instance->m_scriptReporter.AddScreenshotTest(path, envPath);
+        s_instance->m_scriptReporter.AddScreenshotTest(imageName);
 
         s_instance->m_isCapturePending = true;
         s_instance->PauseScript();
 
         return true;
     }
 
+    void ScriptManager::Script_SetScreenshotFolder(const AZStd::string& screenshotFolder)
+    {
+        auto operation = [screenshotFolder]()
+        {
+            AZ::Render::FrameCaptureTestRequestBus::Broadcast(
+                &AZ::Render::FrameCaptureTestRequestBus::Events::SetScreenshotFolder, screenshotFolder);
+        };
+
+        s_instance->m_scriptOperations.push(AZStd::move(operation));
+    }
+
     void ScriptManager::Script_SetTestEnvPath(const AZStd::string& envPath)
     {
-        s_instance->m_envPath = envPath;
+        auto operation = [envPath]()
+        {
+            AZ::Render::FrameCaptureTestRequestBus::Broadcast(
+                &AZ::Render::FrameCaptureTestRequestBus::Events::SetTestEnvPath, envPath);
+        };
```

Comment:
I think this creates a reference to envPath that can cause a use after free. It's probably better to duplicate the string in all the operations.

Reply:
Actually I'm wrong, it's a copy of the referenced type.